torch/nn/_functions/bilinear.py

```diff
@@ -0,0 +1,60 @@
+import torch
+from torch.autograd import Function
+
+
+class Bilinear(Function):
+
+    def forward(self, input1, input2, weight, bias=None):
+        self.save_for_backward(input1, input2, weight, bias)
+
+        output = input1.new()
+
+        buff2 = input1.new()
+        buff2.resize_as_(input2)
+
+        # compute output scores:
+        output.resize_(input1.size(0), weight.size(0))
+        for k, w in enumerate(weight):
+            torch.mm(input1, w, out=buff2)
+            buff2.mul_(input2)
+            torch.sum(buff2, 1, out=output.narrow(1, k, 1))
+
+        if bias is not None:
+            output.add_(bias.expand_as(output))
+
+        return output
+
+    def backward(self, grad_output):
+        input1, input2, weight, bias = self.saved_tensors
+        grad_input1 = grad_input2 = grad_weight = grad_bias = None
+
+        buff1 = input1.new()
```

Comment:
you do not need to use two buffers here. A single one is enough. And as above, you dont need to do `resize_as_` on it.

```
buff = input1.new()
```